File: backend/Domain/zone.go
package domain

import "time"

// Zone represents a geofenced area on campus.
type Zone struct {
	ID          string     `json:"id" bson:"_id"`
	Name        string     `json:"name" bson:"name"`
	Description string     `json:"description" bson:"description"`
	Coordinates []Location `json:"coordinates" bson:"coordinates"` // Polygon points
	RiskLevel   string     `json:"risk_level" bson:"risk_level"`   // "High", "Medium", "Safe"
	Message     string     `json:"message" bson:"message"`         // Message to send upon entry
}

// Notification represents a push notification payload.
type Notification struct {
	ID     string    `json:"id" bson:"_id"`
	UserID string    `json:"user_id,omitempty" bson:"user_id,omitempty"` // Null if broadcast
	Title  string    `json:"title" bson:"title"`
	Body   string    `json:"body" bson:"body"`
	ZoneID string    `json:"zone_id,omitempty" bson:"zone_id,omitempty"` // If triggered by zone
	SentAt time.Time `json:"sent_at" bson:"sent_at"`
}

File: mobile/lib/features/campus_compass/domain/entities/zone.dart
import 'package:equatable/equatable.dart';
import 'package:latlong2/latlong.dart';

class Zone extends Equatable {
  final String id;
  final String name;
  final String description;
  final List<LatLng> coordinates; // Polygon
  final String riskLevel; // 'High', 'Medium', 'Safe'
  final String message;

  const Zone({
    required this.id,
    required this.name,
    required this.description,
    required this.coordinates,
    required this.riskLevel,
    required this.message,
  });

  @override
  List<Object?> get props => [id, name, coordinates, riskLevel];
}
